lib/psm.c

```diff
@@ -469,6 +469,17 @@ static rpmpsm rpmpsmNew(rpmts ts, rpmte te, pkgGoal goal)
 	    psm->scriptArg = npkgs_installed + 1;
 	    psm->countCorrection = 0;
 	    break;
+	case PKG_PREUNTRANS:
+	    if (rpmteDependsOn(psm->te))
```

Comment:
Can't we just use `isUpdate()` here, just like under `case PKG_POSTTRANS`?

Reply:
isUpdate() handles the case where a package updates another, a condition which we need to find out with some difficulty. The erasure case is the reverse: this package was updated by something else, and for that we can just look at the dependent element directly. So no, isUpdate() wouldn't work for this as-is.

Reply:
Oh, that makes sense, thanks for explaining!